### DISCUSSION TITLE
Background image shifts when I show the dialog

### DISCUSSION THREAD

**Discussion by alexander-stolz** [zauberzeug/nicegui#1706] [Q&A] [answered]
### Question

Hello,

I am developing a NiceGUI app and everything works fine so far. The main layout is a left drawer, a background image and a centered column.

First I wanted to achieve this with `ui.image` like this:

```python
with ui.image(...):
  with ui.column(...):
    # content
```

But here the column had a dark transparent background:
![image](https://github.com/zauberzeug/nicegui/assets/4092790/513dcf74-8f81-4b3f-b478-d19e5e658c15)
(1) the background image
(2) the column
(3) one element in the column

The quasar docs didn't help me solving this issue, so I went with another method that worked well:

```python
ui.query(".q-page").style("display: flex; align-items: center;")
ui.query("body").style(
    "background-image: url('/static/backdrop.jpg');"
    "background-size: cover;"
    "background-position: center;"
    "background-repeat: no-repeat;"
    "display: flex;"
    "align-items: center;"
    "justify-content: center;"
)
with ui.column(...):
    # content
```

But now, when I open an `ui.dialog()` the background gets shifted:
![image](https://github.com/zauberzeug/nicegui/assets/4092790/42688feb-cefd-4964-91d6-f137e9447379)
(1) background image
(2) white background, because the background image got shifted
(3) column element

Everything is darker, because the dialog is open (outside of the screenshot).

Does anyone know a solution for my problem?

Thanks!

**Comment by natankeddem** [comment]
I am not sure about your current implementation since you didn't post a full example that I could run. Does something like this do what you want?
```py
from nicegui import ui

with ui.dialog() as dialog, ui.card():
    ui.label('Are you sure?')
    with ui.row():
        ui.button('Yes', on_click=lambda: dialog.submit('Yes'))
        ui.button('No', on_click=lambda: dialog.submit('No'))


async def show():
    result = await dialog
    ui.notify(f'You chose {result}')

with ui.left_drawer(top_corner=True):
    ui.label("drawer")
with ui.image('https://picsum.photos/id/377/640/360'):
    with ui.column():
        ui.label("content")
        ui.button('Await a dialog', on_click=show)


ui.run()
```
Just throwing together a few of the demos from [nicegui.io](https://nicegui.io/)...

**Reply by alexander-stolz** [reply]
Thanks for your reply! I would have posted a working example now, but your code is exactly what I did first (my first pseudo code). But do you see the dark background in your example:

![image](https://github.com/zauberzeug/nicegui/assets/4092790/b729b8a9-f070-42f9-b489-71ada2efa0f0)

I don't want that. This is why I went with the `ui.query('body')` solution that came with it's own caveats.

Here is the code for my solution:

```python
from nicegui import ui

with ui.dialog() as dialog, ui.card():
    ui.label('Are you sure?')
    with ui.row():
        ui.button('Yes', on_click=lambda: dialog.submit('Yes'))
        ui.button('No', on_click=lambda: dialog.submit('No'))


async def show():
    result = await dialog
    ui.notify(f'You chose {result}')


with ui.left_drawer().props("width=180 behavior=desktop"):
    ui.label("drawer")

ui.query(".q-page").style("display: flex; align-items: center;")
ui.query("body").style(
    "background-image: url('https://picsum.photos/id/377/640/360');"
    "background-size: cover;"
    "background-position: center;"
    "background-repeat: no-repeat;"
    "display: flex;"
    "align-items: center;"
    "justify-content: center;"
)
with ui.column():
    ui.label("content")
    ui.button('Await a dialog', on_click=show)


ui.run()
```

**Reply by alexander-stolz** [reply, marked as answer]
Oh.. I just experimented with your code and found a very simple solution:

```python
from nicegui import ui

with ui.dialog() as dialog, ui.card():
    ui.label('Are you sure?')
    with ui.row():
        ui.button('Yes', on_click=lambda: dialog.submit('Yes'))
        ui.button('No', on_click=lambda: dialog.submit('No'))


async def show():
    result = await dialog
    ui.notify(f'You chose {result}')


with ui.left_drawer(top_corner=True):
    ui.label("drawer")
with ui.image('https://picsum.photos/id/377/640/360'):
    with ui.column().style("background: none;").classes("absolute-center"):
        ui.label("content")
        ui.button('Await a dialog', on_click=show)


ui.run()
```
